src/Microsoft.Health.SqlServer/Features/Schema/SchemaJobWorker.cs

```diff
@@ -26,22 +27,30 @@ public class SchemaJobWorker
         private readonly IServiceProvider _serviceProvider;
         private readonly SqlServerDataStoreConfiguration _sqlServerDataStoreConfiguration;
         private readonly IMediator _mediator;
+        private readonly IProcessTerminator _processTerminator;
         private readonly ILogger _logger;
 
         public SchemaJobWorker(
             IServiceProvider services,
             SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration,
             IMediator mediator,
+            IProcessTerminator processTerminator,
             ILogger<SchemaJobWorker> logger)
         {
             EnsureArg.IsNotNull(services, nameof(services));
             EnsureArg.IsNotNull(sqlServerDataStoreConfiguration, nameof(sqlServerDataStoreConfiguration));
             EnsureArg.IsNotNull(mediator, nameof(mediator));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
+            if (sqlServerDataStoreConfiguration.TerminateWhenSchemaVersionUpdatedTo.HasValue)
+            {
+                EnsureArg.IsNotNull(processTerminator, nameof(processTerminator));
+            }
```

Comment:
It doesn't work this way unfortunatelly.

You will get 

Unable to resolve service for type 'Microsoft.Health.Core.Features.Control.IProcessTerminator' while attempting to activate 'Microsoft.Health.SqlServer.Features.Schema.SchemaJobWorker'.


if you try to run it without regestring process terminator


As option, we can have NoopProcessTerminator class which does nothing and register it based on config.